src/pocketmine/Player.php

```diff
@@ -343,6 +343,15 @@ public function setBanned(bool $value){
 		}
 	}
 
+	public function setXUIDBanned(bool $value){
```

Comment:
If this is used on a player who is not logged into Xbox Live, then no Xbox Live players will be able to join.

Reply:
Should I check for $isAuthenticated, then?

Reply:
Wouldn't this just ban all non-xbox live users, since they'd be the only ones with an empty XUID? Checking for isAuthenticated should solve that.